server/backend/src/users/users.service.ts

```diff
@@ -107,6 +152,63 @@ export class UsersService {
     return newCamera;
   }
 
+  /**
+   * Removes all notifications associated with Camera idCam.
+   * @param idCam
+   */
+  public async removeNotifications(idCam: string): Promise<void> {
+    const cam = await this.cameraRepository.findOne(
+      { id: idCam },
+      {
+        populate: [
+          'notifications',
+          'notifications.camera',
+          'notifications.camera.id',
+        ],
+      },
+    );
+
+    if (cam === null) {
+      throw new NotFoundError(`Camera with given ID does not exist.`);
+    }
+
+    cam.notifications.removeAll();
+    await this.cameraRepository.flush();
+  }
+
+  /**
+   * Removes a given camera from one of the user's groups.
+   * @param idCam
+   */
+  public async removeCamera(
+    idUser: string,
+    idGroup: string,
+    idCam: string,
+  ): Promise<boolean> {
+    // add the thing where it checks that the user owns the idCam param.
+
+    const group = await this.groupRepository.findOne(
+      { id: idGroup },
+      { populate: ['cameras'] },
+    );
+
+    const camera = await this.cameraRepository.findOne({ id: idCam });
+
+    if (group === null || camera == null) {
+      throw new NotFoundError(`Camera with given ID does not exist.`);
+    }
+
+    if (camera.group.user.id != idUser) {
+      throw new ForbiddenException();
+    }
+
+    // removes notifications before removing the camera.
+    await this.removeNotifications(idCam);
```

Comment:
Is this step needed if the camera is deleted? Believe that remove-orphans is set to true.